content/docs/guides/parallel-execution.md

```diff
@@ -0,0 +1,380 @@
+		return super.scenarios();
+	}
+}
+```
+{{% /block %}}
+
+{{% block "kotlin" %}}
+```kotlin
+package parallel
+
+import org.testng.annotations.DataProvider;
+import cucumber.api.testng.AbstractTestNGCucumberTests;
+
+class RunCucumberTest : AbstractTestNGCucumberTests() {
+	
+	@DataProvider(parallel = true)
+	override fun scenarios(): Array<Array<(Any)>>  {
+		return super.scenarios()
+	}	
+}
+```
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+- Add the Maven **Surefire plugin configuration** to the `build` section of the `POM`.
+
+```shell
+<plugin>
+	<groupId>org.apache.maven.plugins</groupId>
+	<artifactId>maven-surefire-plugin</artifactId>
+	<version>2.22.0</version>
+</plugin>
+```
+
+- Use the Maven `install` or a suitable command to **execute the POM**. This should run in parallel thread mode. You should see a result similar to below. The **scenarios and rows of the scenario outlines are executed in different threads**.
+
+```shell
+Thread ID - 15 - Scenario Outline Row 2 from scenario-outlines feature file.
+Thread ID - 14 - Scenario Outline Row 1 from scenario-outlines feature file.
+Thread ID - 16 - Scenario 1 from scenarios feature file.
+Thread ID - 17 - [Scenario 2 from scenarios feature file.
+```
+
+- To execute using a Maven **Failsafe plugin**, setup the `POM` as described in the JUnit section. Remove the `parallel` and `useUnlimitedThreads` settings in the `configuration` part.
+
+
+The default **thread count of the dataprovider** in parallel mode is **10**. To change this the `dataproviderthreadcount` property needs to be added to the `configuration` section of the Surefire or Failsafe plugin in the `POM`.
+```shell
+<configuration>
+	<properties>
+    	<property>
+        	<name>dataproviderthreadcount</name>
+        	<value>20</value>
+    	</property>
+	</properties>
+</configuration>
+```
+
+If you have **multiple runners**, set the parallel configuration of `classes` to reduce execution times. In addition the `threadCount` can be set to to the desired value or `useUnlimitedThreads` can be set to true.
+```shell
+<configuration>
+	<parallel>classes</parallel>
+	<threadCount>4</threadCount>
+</configuration>
+```
+{{% /block %}}
+
+# CLI
+
+{{% block "java,kotlin" %}}
+The `Main class` in the `cucumber.api.cli package` is used to execute the feature files. You can run this class directly from the command line; in that case, there is no need to create any runner class. The usage options for this class is mentioned [here](https://github.com/cucumber/cucumber-jvm/blob/v4.0.0/core/src/main/resources/cucumber/api/cli/USAGE.txt). The `--threads` option needs to be set to a value **greater than 1** to run in parallel. When the parallel mode is used, the scenarios and rows in a scenario outline will be run in multiple threads.
+ 
+Below is the basic command to start the execution.
+```shell
+java -cp <classpath> cucumber.api.cli.Main -g <glue package> --threads <thread count> <path to feature files>

```

Comment:
Could you please **also** provide an example of what this command would like look like with the setup as described in this guide (i.e. using directory `parallel`). I'm not too familiar with running Cucumber from the command line so I cannot get this to work.